tooltester/setUpTooltester.sh

```diff
@@ -0,0 +1,149 @@
+  local ENGINE_PROVIDER=$(yq ".contexts.$CONTEXT_NAME.engines[0].engine" "$AGC_CONFIG_FILE")
+  local PROJECT_NAME=$(yq ".name" "$AGC_CONFIG_FILE")
+
+  # Get Batch Compute Environment Name
+  local BATCH_COMPUTE_ENVIRONMENT_NAME=$(aws batch describe-compute-environments --output json | jq ".computeEnvironments[]
+  | select((.computeResources.tags.\"agc-engine\" == \"$ENGINE_PROVIDER\")
+  and (.computeResources.tags.\"agc-context\" == \"$CONTEXT_NAME\")
+  and (.computeResources.tags.\"agc-project\" == \"$PROJECT_NAME\")
+  and (.computeResources.tags.\"agc-engine-type\" == \"$ENGINE_TYPE\")) | .computeEnvironmentName")
+
+  local NUMBER_OF_MATCHING_COMPUTE_ENVIRONMENTS=$(echo "$BATCH_COMPUTE_ENVIRONMENT_NAME" | grep -cv ^$)
+  if [[ $NUMBER_OF_MATCHING_COMPUTE_ENVIRONMENTS -gt 1 ]]
+  then
+    echo "The script has detected more than one matching compute environment"
+    echo "Please set ECS cluster manually using the instructions in README.md"
+    exit 1
+  elif [[ $NUMBER_OF_MATCHING_COMPUTE_ENVIRONMENTS -lt 1 ]]
+  then
+    echo "The script did not detect any matching compute environments"
+    echo "Please ensure that you have started the context $CONTEXT_NAME following the instructions in README.md"
+    echo "You can also set the ECS cluster manually using the instructions in README.md"
+    exit 1
+  fi
+
+  local ECS_CLUSTER_ARN=$(aws ecs list-clusters | jq ".clusterArns[] | select(.|test($BATCH_COMPUTE_ENVIRONMENT_NAME))" | tr -d '"')
+
+  local NUMBER_OF_MATCHING_ECS_ARNS=$(echo "$ECS_CLUSTER_ARN" | grep -cv ^$)
+
+  if [[ $NUMBER_OF_MATCHING_ECS_ARNS -gt 1 ]]
+  then
+    echo "The script has detected more than one matching ECS ARN"
+    echo "Please set the ECS cluster manually using instructions in README.md"
+    exit 1
+  elif [[ $NUMBER_OF_MATCHING_ECS_ARNS -lt 1 ]]
+  then
+    echo "The script did not detect any matching ECS ARNs"
+    echo "Please set the ECS cluster manually using instructions in README.md"
+    exit 1
+  fi
+
+  # Selecting element 0 in array as ECS_CLUSTER_ARN only contains 1 element (verified above), so the below command will
+  # only return 1 result
+  ECS_CLUSTER_NAME=$(aws ecs describe-clusters --cluster "$ECS_CLUSTER_ARN" | jq '.clusters[0].clusterName')
+  ECS_CLUSTER_NAME_NO_QUOTES=$(echo "$ECS_CLUSTER_NAME" | tr -d '"')
+  # Same as above comment
+  # Determines if ECS container insights are enabled, and if they are not enabled, then they are turned on
+  ARE_CONTAINER_INSIGHTS_ENABLED=$(aws ecs describe-clusters --cluster "$ECS_CLUSTER_NAME_NO_QUOTES" --include SETTINGS | jq '.clusters[0].settings[]|select(.name ==  "containerInsights") | .value' | tr -d '"')
+
+  if [[ "$ARE_CONTAINER_INSIGHTS_ENABLED" != "enabled" ]]
+  then
+    echo "We are turing on container insights for this cluster: $ECS_CLUSTER_NAME"
+    aws ecs update-cluster-settings --cluster "$ECS_CLUSTER_NAME_NO_QUOTES" --settings name=containerInsights,value=enabled
```

Comment:
Just an observation, no need to change anything: I'm not sure if everyone will have permissions to do this in AWS

Reply:
Yeah, I do think I needed to get an additional permission to do this. The good news is that if you don't have permission it'll give you a detailed error message saying what permission you need.

It is definitely essential that this command is run (or it is done in console) as without it, we will not have any cloudwatch stats (like `cpuUtilized`).

Reply:
Make a note in the documentation that we settled on the following extra permissions above and beyond what is normally required to run AGC (https://aws.github.io/amazon-genomics-cli/docs/best-practices/iam-permissions/)

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "cloudwatch:DescribeInsightRules",
                "ecs:UpdateCluster",
                "cloudwatch:GetDashboard",
                "ecs:UpdateClusterSettings",
                "cloudwatch:GetInsightRuleReport",
                "cloudwatch:GetMetricData",
                "cloudwatch:GetMetricStatistics",
                "cloudwatch:ListManagedInsightRules",
                "cloudwatch:ListMetrics",
                "cloudwatch:DescribeAnomalyDetectors",
                "cloudwatch:DescribeAlarmHistory",
                "cloudwatch:ListMetricStreams",
                "cloudwatch:DescribeAlarmsForMetric",
                "cloudwatch:ListDashboards",
                "cloudwatch:ListTagsForResource",
                "cloudwatch:DescribeAlarms",
                "cloudwatch:GetMetricStream",
                "cloudwatch:GetMetricWidgetImage"
            ],
            "Resource": "*"
        }
    ]
}
```

Also 
```
ECSReadAccess
AWSCloudShellFullAccess
AmazonS3ReadOnlyAccess
```
were extra as well.

Reply:
Added the required permissions to the PR